### PR TITLE
Use ADO step to test the generated modules

### PR DESCRIPTION
**Pull request by dolauli**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: .azure-pipelines/util/test-steps.yml [hunk 1]
```diff
@@ -16,6 +16,22 @@ steps:
     custom: msbuild
     arguments: 'build.proj /t:${{ parameters.testTarget }} /p:Configuration=${{ parameters.configuration }};TestFramework=${{ parameters.testFramework }};PullRequestNumber=$(System.PullRequest.PullRequestNumber)'
 
+- powershell: |
+   Install-Module -Name Pester -RequiredVersion 4.10.1 -Force -SkipPublisherCheck
+   Install-Module -Name Az.Accounts -Scope CurrentUser -Force
+   Get-ChildItem -Directory Az.* | ForEach-Object {if (Test-Path $_/test-module.ps1) {cd $_; ./test-module.ps1}}
+  workingDirectory: 'artifacts/Debug'
+  displayName: Test for AutoGen Modules With PowerShell
```
Comment (dingmeng-xue) [inline .azure-pipelines/util/test-steps.yml:24]: With .NET Framework

File: .azure-pipelines/util/test-steps.yml [hunk 2]
```diff
@@ -16,6 +16,22 @@ steps:
...
+  condition: and(succeeded(), ne('${{ parameters.testFramework }}', 'netcoreapp2.2'))
+
+- pwsh: |
+   Install-Module -Name Pester -RequiredVersion 4.10.1 -Force
+   Install-Module -Name Az.Accounts -Scope CurrentUser -Force
+   Get-ChildItem -Directory Az.* | ForEach-Object {if (Test-Path $_/test-module.ps1) {cd $_; ./test-module.ps1}}
+  workingDirectory: 'artifacts/Debug'
+  displayName: 'Test for AutoGen Modules With PowerShell Core'
```
Comment (dingmeng-xue) [inline .azure-pipelines/util/test-steps.yml:32]: With PowerShell Core or PowerShell 7

File: .azure-pipelines/util/test-steps.yml [hunk 3]
```diff
@@ -16,6 +16,22 @@ steps:
     custom: msbuild
     arguments: 'build.proj /t:${{ parameters.testTarget }} /p:Configuration=${{ parameters.configuration }};TestFramework=${{ parameters.testFramework }};PullRequestNumber=$(System.PullRequest.PullRequestNumber)'
 
+- powershell: |
+   Install-Module -Name Pester -RequiredVersion 4.10.1 -Force -SkipPublisherCheck
+   Install-Module -Name Az.Accounts -Scope CurrentUser -Force
+   Get-ChildItem -Directory Az.* | ForEach-Object {if (Test-Path $_/test-module.ps1) {cd $_; ./test-module.ps1}}
+  workingDirectory: 'artifacts/Debug'
+  displayName: Test for AutoGen Modules With PowerShell
+  condition: and(succeeded(), ne('${{ parameters.testFramework }}', 'netcoreapp2.2'))
```
Comment (dingmeng-xue) [inline .azure-pipelines/util/test-steps.yml:25]: TestFramework: net472
  TestTarget: TestNet472

**Pull request merged**